### PR TITLE
5pm c avg class size by dept

### LINKED ISSUE #46: Average class size by department
# User Story

As a user I can find the average class size(not occupancy) within a specified start and end quarter so that we can view the statistics across multiple departments(Including Undergraduate and Graduate Courses).

# Acceptance Criteria

- [x] In the Statistics dropdown menu, there should be an entry that links to a page that allows you to see the average class size by department.
- [x] There is a form on the average class size page that lets you select a quarter range.
- [x] After submitting the form, it should display average class sizes across all departments.
...

# Implementation Todos

Front end:

- [x] Create a ClassSize.js page that is linked in the Statistics Navbar
- [x] Add a form that allows the user to select a quarter range
- [x] After selecting the quarter range, the results appear in a neatly formed table 
...

Back end:

- [x] Endpoint to retrieve occupancy information given a department, start and end quarters
...

Testing:

- [x] Front end tests pass and there is adequate coverage
- [x] Back end tests pass and there is adequate coverage
...

### PR DESCRIPTION
**Pull request by paritoshjha24**
This PR implements the "Average Class Size by Department" statistics page, where a user can select a start and end quarter and view the average class size for all departments (including Undergraduate and Graduate courses) over that range of time. Closes #46 

Before:

course-search-pr-76-before

After: 

course-search-pr-76-after

### CODE REVIEW

File: javascript/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -44,9 +44,12 @@ function AppNavbar() {
           <NavDropdown.Item as={Link} to="/statistics">
             Course Occupancy by Class Level
           </NavDropdown.Item>
-          <NavDropdown.Item as={Link} to="/statistics">
+          <NavDropdown.Item as={Link} to="/statistics/classSize">
             Average Class Size by Department
           </NavDropdown.Item>
+          <NavDropdown.Item as={Link} to="/statistics/courseCount">
```
Comment (jvoucsb) [inline javascript/src/main/components/Nav/AppNavbar.js:50]: I don't think addition of this courseCount nav drop down item is valid. Consider removing

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,4 @@
 spring.config.location=classpath:/
 @environment-properties@
+
+server.port=${PORT:8080}
```
Comment (jvoucsb) [inline src/main/resources/application.properties:4]: Should this be included in the changes?

**Review by jvoucsb — CHANGES_REQUESTED** [review]
Deployed locally and found no bugs.

There is one change needed and one to consider. Other than that, looks good to me.

File: src/main/resources/application.properties (hunk 2 above)
Comment (paritoshjha24) [reply src/main/resources/application.properties:4]: This line saves the user a little time and research when port forwarding. With this line, user can run 
"PORT=XXXX mvn spring:boot-run" to port forward

**Review by hdinh77 — APPROVED** [review]
besides the changes about PORT that justin mentioned (this is more for a local user that needs to use a different port), looks good!

File: javascript/src/main/App.js [hunk 3]
```diff
@@ -40,7 +41,11 @@ function App() {
           <Route path="/history/basic" exact component={Basic} />
           <Route path="/history/courseName" exact component={CourseName} />
           <Route path="/statistics" exact component={Statistics} />
+
+          <Route path="/statistics/classSize" exact component={ClassSize} />
```
Comment (jvoucsb) [inline javascript/src/main/App.js:45]: Just some unnecessary space here

**Review by jvoucsb — COMMENTED** [review]
Quick fix. But otherwise, looks good to me.

File: javascript/src/main/App.js (hunk 3 above)
Comment (pconrad) [reply javascript/src/main/App.js:45]: Good catch... But under the circumstances I can live with this :-)

**Review by phtcon — APPROVED** [review]
LGTM.  40 points

**Pull request merged**